### DISCUSSION TITLE
How to Stop Using Teslamate for a while

### DISCUSSION THREAD

**Discussion by JerinSaji0** [teslamate-org/teslamate#1417] [Q&A]
Hello,

I was just told by Tesla Service Center that Teslamate might be the reason why I've encountered some Autopilot & Infotainment screen issues recently. How do I safely revoke API access for Teslamate to see if their theory is true or not without having to setup TeslaMate again when I'm good to use it again?

**Comment by rmyadsk** [comment]
Log out of Teslamate, then stop the Teslamate container/server. Then, change your Tesla account password - that should revoke all existing API tokens that are out in the wild. (All other third party apps should cease functioning as well.)

Though I'm quite curious why they suggested Teslamate might be causing some issues! It uses their own well-defined and well-documented API readonly.

**Reply by JerinSaji0** [reply]
Done. Thank you!

I was getting the " Autopilot safety/convenience features unavailable/Features may be restored on next drive" error in the middle of a roadtrip and green infotainment malfunction when the rear camera was on. They said it might be because of this but it sounded like they don't know themselves. I work in Cybersecurity so I can't think of any way how a readonly API could cause this but I'm just going to take the safe route and stop using TeslaMate for the time-being. If this happens again, call them out on their bs. Driving 600mi without even basic autopilot was not fun when you paid $10k for FSD....

**Reply by wooter** [reply]
Just FYI: resetting the password of your Tesla account to the same password is enough. It expires all open tokens used by 3rd party apps and services, including TeslaMate.

**Reply by ngardiner** [reply]
Tesla often suggest this. Their API isn't publicly documented, it was reverse engineered from the app and Tesla certainly isn't developer friendly. I'd take any suggestion of Tesla's that a 3rd party with a bowl of salt, but I'd also point out that Tesla merely tolerate the use of the API at this point and people using TeslaMate or similar tools should probably turn them off when they're having issues and not mention them to Tesla lest we get another round of API blocking or WAF related connectivity issues to filter out said 3rd party tools.

**Comment by brianmay** [comment]
If Teslamate is causing issues, you can stop Teslamate. No need to revoke API access (unless you have fears that something else might be accessing the API).

Tesla in my experience is very anti-third party and will try to blame any issues on 3rd party components (hardware and software), even when there is no actual evidence.

But consider: This API is the same API that is used by the mobile app (might have changed now but I am skeptical).

Unfortunately, other car brands aren't any better. I see that there has been work to reverse engineer the MG protocols. But not officially supported, so it could stop working. BYD seems to be getting nowhere.

Maybe installing our own logger might be a better approach. Google shows me links like https://docs.openvehicles.com/en/latest/introduction.html https://github.com/bassmaster187/TeslaLogger - I seem to remember this being discussed here previously, with a reference to a good OVMS adapter that I can't find right now. But now Tesla is talking about locking down the API more and more - particularly for older cars which have limited support with the newer APIs this might be worth revisiting.

Having a charger that supports OCPP might be a better bet for turning on/off charging. Although I have no idea what OCPP is like to program, and what data - if any - it gives you. There is rumors that Tesla Wall chargers might get OCPP in the future, but I won't believe it until it happens.

I suspect the reversed engineered API is not going away anytime soon.  I think Tesla doesn't really care enough. They just want another excuse to charge large fleet owners $$$ on a subscription basis, and they have got that. But the fact it could disappear without warning and break my systems that rely on streaming, and there is no newer streaming API that works with the older cars, kind of reduces my enthusiasm for developing this code.

**Reply by brianmay** [reply]
Oops. May bad. Somehow got confused and responded to a post from 2021.

**Reply by malyjaponec** [reply]
No problem. I think it is still issue. I have nissan leaf with OVMS and if there is something like teslamate that translate data from OVMS into grafana/database i would love it. It can pust data to server itself. I am using it with app and free/donated server but the OVMS unit can send data to 2 servers (it has 2 different protocols) but I have installed also simple script in language like java script that push data into ABRP. It is very easy to send data to anything. So if "teslamate" dockered system is equiped with some kind of receiver paralel to teslamate container it could with OVMS handle many cars. OVMS is still evolving as it is open project and people adding support to new cars. Together woudl be very nice tool. I have all telemetry from leaf every 5 seconds but I have no tool that database it and graphically display.
